src/main.c

```diff
@@ -235,13 +225,71 @@ static void check_for_and_print_sys_message(const char* fmt, ...) {
   
 }
 
-#define INFO(fmt, ...)  check_for_and_print_sys_message(fmt, ##__VA_ARGS__); \
+static int index_of_string_limited(const char *str, int len, const char *search_string, int start_pos, int search_limit){
+  int search_len = strlen(search_string);
+  int last_possible_start = len < search_limit ? len - search_len : search_limit - search_len;
+  int pos = start_pos;
+
+  if (start_pos > last_possible_start){
+    return -1;
+  }
+  while (pos <= last_possible_start){
+    if (!memcmp(str+pos,search_string,search_len)){
+      return pos;
+    }
+    pos++;
+  }
+  return -1;
+}
+
+//size of "ZWE_zowe_sysMessages"
+#define ZWE_SYSMESSAGES_EXCLUDE_LEN 20
+
+// matches YYYY-MM-DD starting with 2xxx.
+// this regex was chosen because other patterns didnt seem to work with LE's regex library.
+#define DATE_PREFIX_REGEXP_PATTERN "^[2-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9].*"
+
+// zowe standard "YYYY-MM-DD HH-MM-SS.sss "
+#define DATE_PREFIX_LEN 24
+
+static void check_for_and_print_sys_message(const char* input_string) {
+  if (!zl_context.sys_messages) {
+    return;
+  }
+
+  int count = jsonArrayGetCount(zl_context.sys_messages);
+  int input_length = strlen(input_string);
+  for (int i = 0; i < count; i++) {
+    const char *sys_message_id = jsonArrayGetString(zl_context.sys_messages, i);
+    if (sys_message_id && (index_of_string_limited(input_string, input_length, sys_message_id, 0, SYSLOG_MESSAGE_LENGTH_LIMIT) != -1)) {
+
+      //exclude "ZWE_zowe_sysMessages" messages to avoid spam.
```

Comment:
With this you can remove the need for `ZWE_SYSMESSAGES_EXCLUDE_LEN` too

Reply:
2023317 is the current date in julian format, as that is what SYSLOG has always used for dates, which converts to 2023-11-13.  Whereas the "example line" above it is from the joblog output which does NOT have a date preceding it.

Reply:
i think applyEnviron is a debug-level message (zowe logLevel=debug/trace) so i didn't catch that but it would be good to remove.
i also note that the app-server prints out the env vars too, in a way that isnt caught by this code and shouldnt be. it would be better to make a pr in app-server to omit them.